device/core/test/_internal_client_test.js

```diff
@@ -790,6 +873,27 @@ var ClientPropertyCollection = require('../dist/pnp/properties').ClientPropertyC
         assert.deepEqual(twin._updateReportedProperties.args[0][0], {key: "value"});
       });
 
```

Comment:
I don't see any tests where you pass a first parameter to get that ISN'T a function or a boolean type.